File: Backend/Application/Groups/Queries/GroupDetail/GroupUserViewModel.cs
using Domain.Models;

namespace Application.Groups.Queries.GroupDetail
{
    public class GroupUserViewModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public bool IsManager { get; set; }
        public bool IsOwner { get; set; }
        public bool IsAcceptedByManager { get; set; }
        public bool IsAcceptedByUser { get; set; }
        public string RequestMessage { get; set; }

        public GroupUserViewModel(UserGroup userGroup)
        {
            Id = userGroup.UserId;
            Name = userGroup.User.Username;
            IsManager = userGroup.IsManager == true;
            IsOwner = userGroup.Group.OwnerId == Id;
            IsAcceptedByManager = userGroup.IsAcceptedByManager == true;
            IsAcceptedByUser = userGroup.IsAcceptedByUser == true;
            RequestMessage = userGroup.RequestMessage;
        }
    }
}

File: Frontend/models/getter/group/GroupUserViewModel.ts
export default class GroupUserViewModel {
  public Id: number;
  public Name: string;
  public IsManager: boolean;
  public IsOwner: boolean;
  public IsAcceptedByUser: boolean;
  public IsAcceptedByManager: boolean;
  public RequestMessage: string;

  constructor(data?) {
    if (!data) return;
    this.Id = data.id;
    this.Name = data.name;
    this.IsManager = data.isManager;
    this.IsOwner = data.isOwner;
    this.IsAcceptedByUser = data.isAcceptedByUser;
    this.IsAcceptedByManager = data.isAcceptedByManager;
    this.RequestMessage = data.requestMessage;
  }
}
